### ISSUE TITLE
Forgot password flow fails to complete; dialog is stuck and won't process clicks

### ISSUE THREAD

**Issue by jrgm** [mozilla/persona#1440]
Forgot password flow fails to complete; dialog is stuck and won't process clicks

Steps to reproduce:
1. Go to http://beta.myfavoritebeer.org and click signin.
2. If email addresses are listed, click 'This is not me'.
3. In the email textbox, enter an email address for a "non-primary" account that is already known to diresworb.org, and click Next.
4. When prompted for a password, click the 'Forgot your password?' link.
5. Click 'reset password'.
6. Fetch the email, and load the /verify_email_address in the same browser.
7. The /sign_up dialog now transitions to screen with your email address, and a 'sign in' button.

Expected:
Clicking 'sign in' will complete sign in.

Actual:
Nothing happens click you click 'sign in'.

Notes:
- I don't see any errors in the js console.
- I see no network traffic when clicking the 'sign in' button.
- Note that the links for 'Add another email' and 'This is not me' do not work either (although the plain links for TOS and PP do work).
- The user's only option is to quit the dialog and try to sign in again (which will work, presenting the 'is this your computer' screen).
- reproduced with OSX/FF, OSX/Chrome, WinXP/IE8, Win7/IE9

Gonna mark this 5-star, and we can discuss the real severity/priority tomorrow. (Actually, this flow doesn't work in a different way with production).

**Comment by lloyd** [comment]
@shane-tomlinson lemme know if you want any help on this one!  /cc @ozten

**Comment by shane-tomlinson** [comment]
@jrgm, @lloyd - I see two problems occurring here.  The first is the backend sometimes returns "mustAuth" for "user_creation_status", even when completing registration in the same browser as beginning the flow.  The second is the front end is not handling the "mustAuth" case correctly.

Important info:

1) I signed in to diresworb.org using set117+test@gmail.com
2) I completed the registration in the same browser as I started the flow.

Request/Response info for https://diresworb.org/wsapi/user_creation_status?email=set117%2Btest%40gmail.com
The cookie has been removed for clarity.

Params:

```
email   set117+test@gmail.com
```

Headers:

Response Headers

```
Cache-Control   no-cache, max-age=0
Connection  keep-alive
Content-Length  21
Content-Type    application/json; charset=utf-8
Date    Tue, 17 Apr 2012 10:47:22 GMT
Set-Cookie  browserid_state_6f1f7c=biglongcookie; path=/wsapi; expires=Tue, 01 May 2012 10:47:22 GMT; secure; httponly
Strict-Transport-Security   max-age=2592000; includeSubdomains
x-frame-options DENY
```

Request Headers

```
Accept  application/json;text/plain
Accept-Encoding gzip, deflate
Accept-Language en,en-us;q=0.8,es-es;q=0.7,db-lb;q=0.5,en-gb;q=0.3,chrome://global/locale/intl.properties;q=0.2
Connection  keep-alive
Content-Type    application/x-www-form-urlencoded
Cookie  browserid_state_6f1f7c=biglongcookie
Host    diresworb.org
Referer https://diresworb.org/sign_in
User-Agent  Mozilla/5.0 (Macintosh; Intel Mac OS X 10.7; rv:11.0) Gecko/20100101 Firefox/11.0
X-Requested-With    XMLHttpRequest
```

Response:

```
{"status":"mustAuth"}
```

I have gone through this flow 3 times.  2 out of 3 times, I received a response of mustAuth.

**Comment by lloyd** [comment]
we're going to split this into two issues.  the first is this one - the fact that reset_password is completely broken and button doesn't respond to clicks.  the second is the race condition whereby sometimes you'll needlessly be required to re-enter your password after email verification - this will be a new issue.

**Comment by lloyd** [comment]
merged into train-2012.04.11 and into dev.

**Comment by jrgm** [comment]
Working for me on stage train-2012.04.11-7. Continuing with some other browsers, etc.

**Comment by jrgm** [comment]
verified fixed for stage train-2012.04.11-8 on multiple browsers.
